vllm/model_executor/sampling_metadata.py

```diff
@@ -432,13 +433,15 @@ def from_sampling_metadata(
                 if (seq_group.is_prompt
                         and sampling_params.prompt_logprobs is not None):
                     prefill_len = len(seq_group.prompt_logprob_indices)
-                    prompt_tokens.extend([] for _ in range(prefill_len))
-                    output_tokens.extend([] for _ in range(prefill_len))
+                    prompt_tokens.extend(
+                        array('l') for _ in range(prefill_len))
+                    output_tokens.extend(
+                        array('l') for _ in range(prefill_len))
```

Comment:
Is there any reason to choose "signed long" here? Maybe unsigned would make more sense for token ids. Reference: https://docs.python.org/3/library/array.html
Also might it be more efficient to make one list and use it twice in these extends?

Reply:
Because the code base already use `torch.int64` and `torch.long`.

https://github.com/vllm-project/vllm/blob/d9c591e9544c2d8199f50e62fb904c28405d317e/vllm/model_executor/sampling_metadata.py#L470-L488

This is furthermore due to PyTorch's limited support for unsigned int.

Reply:
Makes sense, thank you